File: manual/data/cpp/src/function_catboost_params.cpp
#include <boost/property_tree/ptree.hpp>
#include <boost/lexical_cast.hpp>
#include <string>
#include <sstream>

class BoostParamFetcher {
public:
    std::string get_model_params(int iterations, int depth, float learning_rate) {
        boost::property_tree::ptree model_params;
        model_params.put("iterations", iterations);
        model_params.put("depth", depth);
        model_params.put("learning_rate", learning_rate);

        std::stringstream result;
        result << "Model Params - "
               << "Iterations: " << model_params.get<int>("iterations") << ", "
               << "Depth: " << model_params.get<int>("depth") << ", "
               << "Learning Rate: " << model_params.get<float>("learning_rate");

        return result.str();
    }
};

File: manual/data/python/src/function_catboost_params.py
from catboost import CatBoostClassifier


class CatBoostParamFetcher:
    def get_model_params(self, iterations: int, depth: int, learning_rate: float) -> str:
        model = CatBoostClassifier(iterations=iterations, depth=depth, learning_rate=learning_rate, silent=True)
        model_params = model.get_params()
        iterations = model_params["iterations"]
        depth = model_params["depth"]
        learning_rate = model_params["learning_rate"]

        return f"Model Params - Iterations: {iterations}, Depth: {depth}, Learning Rate: {learning_rate}"
